### PR TITLE
python311Packages.tensorflow: mark as broken

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

Tensorflow does not build on python 3.11.

cc @abbradar 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by samuela** [comment]
Why does TF not build on python 3.11? is there an issue with our build of TF or is it not a supported python version upstream?

**Comment by GaetanLepage** [comment]
> Why does TF not build on python 3.11? is there an issue with our build of TF or is it not a supported python version upstream?

No idea... I don't think that it is incompatible with 3.11.
For me it is a compile failure that needs further investigation.

**Comment by samuela** [comment]
Ok that's the sort of thing that we should leave a comment about with a link to the complete failure log. Otherwise this will become mystery code

File: pkgs/development/python-modules/tensorflow/default.nix [hunk 1]
```diff
@@ -43,8 +43,8 @@ let
   # 2. Normal C++ files should be compiled with the same toolchain,
   #    to avoid potential weird dynamic linkage errors at runtime.
   #    This may not be necessary though
-  #
   # Toolchain for Darwin:
+  #
```
Comment (samuela) [inline pkgs/development/python-modules/tensorflow/default.nix:47]: why is this change included?

**Comment by dotlambda** [comment]
> Ok that's the sort of thing that we should leave a comment about with a link to the complete failure log. Otherwise this will become mystery code

We don't know why it fails, just that it's supposed to work. That's why we use `broken`, not `disabled`. I think that's enough of a comment given that you can just try to build it to see whether it was fixed.

**Review by dotlambda — APPROVED** [review]
